### PR TITLE
Reactor fixes in mailbox cassandra

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[c9c5ccb2a5bcb3b86e9904ec214d703f3f89bb5c] [ERROR] Errors: 
[c9c5ccb2a5bcb3b86e9904ec214d703f3f89bb5c] [ERROR]   WebAdminUtilsTest.serverShouldBeAbletoStartConcurrently:38 » NotTerminated
```
test this please

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraUidProvider.java [hunk 1]
```diff
@@ -100,14 +102,15 @@ public MessageUid nextUid(MailboxId mailboxId) throws MailboxException {
...
+        Duration forever = Duration.ofMillis(Long.MAX_VALUE);
         Mono<MessageUid> updateUid = findHighestUid(cassandraId)
             .flatMap(messageUid -> tryUpdateUid(cassandraId, messageUid));
 
         return updateUid
             .switchIfEmpty(tryInsert(cassandraId))
             .switchIfEmpty(updateUid)
             .single()
-            .retry(maxUidRetries);
+            .retryBackoff(maxUidRetries, Duration.ofMillis(10), forever, Schedulers.elastic());
```
Comment (remk) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraUidProvider.java]: Make a constant of the backoff first delay ?

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraModSeqProvider.java [hunk 2]
```diff
@@ -188,7 +188,7 @@ public ModSeq highestModSeq(MailboxId mailboxId) throws MailboxException {
         Duration forever = Duration.ofMillis(Long.MAX_VALUE);
         return tryFindThenUpdateOnce(mailboxId)
             .single()
-            .retryBackoff(maxModSeqRetries, Duration.ofMillis(2), forever, Schedulers.elastic());
+            .retryBackoff(maxModSeqRetries, Duration.ofMillis(10), forever, Schedulers.elastic());
```
Comment (remk) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraModSeqProvider.java]: idem

**Comment by Arsnael** [comment]
```
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR] Errors: 
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR]   WebAdminUtilsTest.serverShouldBeAbletoStartConcurrently:38 » NotTerminated
```
test this please

**Comment by mbaechler** [comment]
```
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [INFO] Running org.apache.james.blob.memory.MemoryDumbBlobStoreTest
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR] Tests run: 64, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 14.631 s <<< FAILURE! - in org.apache.james.blob.memory.MemoryDumbBlobStoreTest
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR] mixingSaveReadAndDeleteShouldReturnConsistentState  Time elapsed: 5.624 s  <<< ERROR!
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] Caused by: java.lang.OutOfMemoryError: Java heap space
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] 
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [INFO] Running org.apache.james.blob.memory.MemoryBlobStoreTest
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [INFO] Tests run: 77, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.057 s - in org.apache.james.blob.memory.MemoryBlobStoreTest
```

test this please

**Comment by mbaechler** [comment]
```
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR] Errors: 
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR]   RemoteDeliveryDKIMIntegrationTest$WhenEnable8BitMime.remoteDeliveryShouldNotBreakDKIMSign:205 » PermFail
```

test this please

**Comment by mbaechler** [comment]
```

[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR] Errors: 
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR]   MemorySpamAssassinContractTest.deletingSpamMessageShouldNotImpactSpamAssassinState » ConditionTimeout
```

test this please

**Comment by mbaechler** [comment]
```
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR] Errors: 
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR]   MemorySpamAssassinContractTest.movingAMailToTrashShouldNotImpactSpamassassinLearning » ConditionTimeout
```
Either no luck or something is broken

test this please

**Comment by Arsnael** [comment]
tests are :green_apple: 

```
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR] Errors: 
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR]   JamesDeploymentValidationTest>DeploymentValidation.validateDeploymentWithMailsFromSmtp:95 » ConditionTimeout
```
test this please

**Comment by mbaechler** [comment]
```
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR] Errors: 
[3f911092609033f6b6f2c48b4a58ce4a8f5e84d4] [ERROR]   WebAdminUtilsTest.serverShouldBeAbletoStartConcurrently:38 » NotTerminated
```

test this please

**Comment by mbaechler** [comment]
Merged

**Pull request closed without merging**